core/services/src/main/java/org/exoplatform/services/attachments/storage/AttachmentStorageImpl.java

```diff
@@ -16,42 +16,112 @@
+    this.documentService = documentService;
+    this.linkManager = linkManager;
   }
 
   @Override
-  public void linkAttachmentsToEntity(long entityId, String entityType, List<String> attachmentsIds) {
-    attachmentsIds.forEach(attachmentId -> {
-      AttachmentContextEntity attachmentContextEntity = new AttachmentContextEntity();
-      attachmentContextEntity.setEntityId(entityId);
-      attachmentContextEntity.setEntityType(entityType.toUpperCase());
-      attachmentContextEntity.setAttachmentId(attachmentId);
-      attachmentContextEntity.setAttachedDate(System.currentTimeMillis());
-      attachmentDAO.create(attachmentContextEntity);
-    });
-
+  public void linkAttachmentToEntity(long entityId, String entityType, String attachmentId) {
+    AttachmentContextEntity attachmentContextEntity = new AttachmentContextEntity();
+    attachmentContextEntity.setEntityId(entityId);
+    attachmentContextEntity.setEntityType(entityType.toUpperCase());
+    attachmentContextEntity.setAttachmentId(attachmentId);
+    attachmentContextEntity.setAttachedDate(System.currentTimeMillis());
+    attachmentDAO.create(attachmentContextEntity);
   }
 
   @Override
-  public List<AttachmentContextEntity> getAttachmentContextByEntity(long entityId, String entityType) {
-    return attachmentDAO.getAttachmentContextByEntity(entityId, entityType.toUpperCase());
+  public List<Attachment> getAttachmentsByEntity(long userIdentityId,
+                                                 long entityId,
+                                                 String entityType) throws ObjectNotFoundException {
+    List<AttachmentContextEntity> attachmentsContextEntity = attachmentDAO.getAttachmentContextByEntity(entityId,
+                                                                                                        entityType.toUpperCase());
+    Utils.sortAttachmentsByDate(attachmentsContextEntity);
+    List<Attachment> attachments = new ArrayList<>();
+    if (!attachmentsContextEntity.isEmpty()) {
+      Session session = null;
+      try {
+        session = Utils.getSession(sessionProviderService, repositoryService);
+        for (AttachmentContextEntity attachmentContextEntity : attachmentsContextEntity) {
+          Attachment attachment = EntityBuilder.fromAttachmentNode(repositoryService,
+                                                                   documentService,
+                                                                   linkManager,
+                                                                   Utils.getCurrentWorkspace(repositoryService),
+                                                                   session,
+                                                                   attachmentContextEntity.getAttachmentId());
+          attachments.add(attachment);
+        }
+      } catch (Exception e) {
+        throw new ObjectNotFoundException("Can't convert attachment JCR node to Attachment DTO");
```

Comment:
not compatible exceptions, can't transform all `Exception` => `ObjectNotFoundException`